source/server/server.cc

```diff
@@ -1091,7 +1091,13 @@ void InstanceBase::notifyCallbacksForStage(Stage stage, std::function<void()> co
   ASSERT_IS_MAIN_OR_TEST_THREAD();
   const auto it = stage_callbacks_.find(stage);
   if (it != stage_callbacks_.end()) {
-    for (const StageCallback& callback : it->second) {
+    LifecycleNotifierCallbacks& callbacks = it->second;
+    for (auto it = callbacks.begin(); it != callbacks.end();) {
```

Comment:
can we have the inner and outer iterators be different names? I realize this is legal but it's confusing :)

Reply:
Done